dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -359,10 +372,18 @@ Path findClassLocation(String className) {
    */
   private ImmutableSet<JavaClass> listClassesInJar(Path jar) throws IOException {
     ImmutableSet.Builder<JavaClass> javaClasses = ImmutableSet.builder();
+
+    // Not to keep JavaClasses retained by class repository while parsing hundreds of jar files
+    Repository classRepository = createClassRepository(inputClassPath);
     for (String className : listClassNamesInJar(jar)) {
```

Comment:
Again, I think you're dealing with a path rather than a name. Distinguishing that and adding some comments should make it clearer what this is doing.

Reply:
It's a path separated by "." (usually it's same as name)